### ISSUE TITLE
tuist cache warm adds macOS frameworks to iOS target

### ISSUE THREAD

**Issue by ilia3546** [tuist/tuist#5647]
### What happened?

[Firebase package file](https://github.com/firebase/firebase-ios-sdk/blob/master/Package.swift) contains following linkerSettings:

```swift
linkerSettings: [
        .linkedFramework("UIKit", .when(platforms: [.iOS, .macCatalyst, .tvOS])),
        .linkedFramework("AppKit", .when(platforms: [.macOS])),
]
```

If we add firebase to our project, and exec `$ tuist generate` then AppKit framework linked to App target correctly:

<img width="500" alt="Correct linked framework" src="https://github.com/tuist/tuist/assets/4445510/dd34cf4f-b72a-484e-9127-91d52947dbb7">

As you can see AppKit used only if we use macOS platform.

If we run `$ tuist cache warm --dependencies-only` and then generate project, we can see that AppKit framework not linked to App target correctly:

<img width="500" alt="Not correct linked framework" src="https://github.com/tuist/tuist/assets/4445510/f5e27435-9fc6-408e-87ed-aa26f95b76ce">

As you can see AppKit used both MacOS and iOS platforms.

If we try to build target for iOS platform we will catch the following error:

```sh
ld: framework 'AppKit' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### How do we reproduce it?

1. Take any project with Firebase dependency
2. Run `tuist cache warm --dependencies-only`
3. Run `tuist generate`
4. Try to build app target
5. Catch error

### Error log

```sh
ld: framework 'AppKit' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### macOS version

14.12

### Tuist version

3.35.1

### Xcode version

15.0.1

**Comment by stefanomondino** [comment]
Just hit same issue here. I'm afraid only workaround (for now) can be disabling cache warming or revert back to previous versions when multiplatform was still unsupported

**Comment by muukii** [comment]
same here

**Comment by kapitoshka438** [comment]
That's annoying bug that needs to be solved asap gyus.
As a workaround for now we have to remove AppKit.framework manually this way:
`sed -i '' '/AppKit.framework/d' {path-to-generated-bpxproj}`

**Comment by pepicrft** [comment]
This bug should have been fixed in 3.36.1. Would you mind updating and letting us know if the issue goes away?

**Comment by ilia3546** [comment]
> This bug should have been fixed in 3.36.1. Would you mind updating and letting us know if the issue goes away?

In 3.36.1 after exec `tuist cache warm --dependencies-only` I catch the following error:

```
xcodebuild: error: Scheme ProjectCache-Binaries-iOS is not currently configured for the build action.
```

**Comment by pepicrft** [comment]
Hey @ilia3546 👋🏼 
We noticed the same regression. We are looking into a fix for it :)

**Comment by kapitoshka438** [comment]
> This bug should have been fixed in 3.36.1. Would you mind updating and letting us know if the issue goes away?

The issue hasn't gone away in 3.36.1

**Comment by fortmarek** [comment]
As reported [here](https://github.com/tuist/tuist/issues/5796#issuecomment-1892771952), this issue should be fixed now.
